include/linux/blkdev_compat.h

```diff
@@ -311,10 +311,14 @@ bio_set_flags_failfast(struct block_device *bdev, int *flags)
  * the block layers responsibility to choose the correct way to
  * implement these semantics.
  */
-#ifdef WRITE_FLUSH_FUA
+#if defined(WRITE_FLUSH_FUA)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_FLUSH_FUA
-#else
+#elif defined(WRITE_BARRIER)
 #define	VDEV_WRITE_FLUSH_FUA		WRITE_BARRIER
+#elif defined(HAVE_REQ_FUA)
+#define	VDEV_WRITE_FLUSH_FUA		REQ_FUA
```

Comment:
So it looks like we can use REQ_PREFLUSH for >= 4.10 kernels.  What about the priority of flags for older kernels?    Prior to 4.10, it looks like REQ_FLUSH will work back to 2.6.36 and then before that, it looks like we could use either WRITE_BARRIER or REQ_SOFTBARRIER.  Should the precedence then be:
 - REQ_PREFLUSH
 - REQ_FLUSH
 - REQ_SOFTBARRIER (or stick with WRITE_BARRIER?)